service/account/providers/gcp/gcpcredentials.go

```diff
@@ -0,0 +1,216 @@
+	ClientEmail             *string `json:"client_email,omitempty"`
+	ClientId                *string `json:"client_id,omitempty"`
+	AuthUri                 *string `json:"auth_uri,omitempty"`
+	TokenUri                *string `json:"token_uri,omitempty"`
+	AuthProviderX509CertUrl *string `json:"auth_provider_x509_cert_url,omitempty"`
+	ClientX509CertUrl       *string `json:"client_x509_cert_url,omitempty"`
+
+	forceSendFields []string
+
+	nullFields []string
+}
+
+func (o ServiceAccounts) MarshalJSON() ([]byte, error) {
+	type noMethod ServiceAccounts
+	raw := noMethod(o)
+	return jsonutil.MarshalJSON(raw, o.forceSendFields, o.nullFields)
+}
+
+func (o *ServiceAccounts) SetAccountId(v *string) *ServiceAccounts {
+	if o.AccountId = v; o.AccountId == nil {
+		o.nullFields = append(o.nullFields, "AccountId")
+	}
+	return o
+}
+
+func (o *ServiceAccounts) SetType(v *string) *ServiceAccounts {
+	if o.Type = v; o.Type == nil {
+		o.nullFields = append(o.nullFields, "Type")
+	}
+	return o
+}
+
+func (o *ServiceAccounts) SetProjectId(v *string) *ServiceAccounts {
+	if o.ProjectId = v; o.ProjectId == nil {
+		o.nullFields = append(o.nullFields, "ProjectId")
+	}
+	return o
+}
+
+func (o *ServiceAccounts) SetPrivateKeyId(v *string) *ServiceAccounts {
+	if o.PrivateKeyId = v; o.PrivateKeyId == nil {
+		o.nullFields = append(o.nullFields, "PrivateKeyId")
+	}
+	return o
+}
+
+func (o *ServiceAccounts) SetPrivateKey(v *string) *ServiceAccounts {
+	if o.PrivateKey = v; o.PrivateKey == nil {
+		o.nullFields = append(o.nullFields, "PrivateKey")
+	}
+	return o
+}
+
+func (o *ServiceAccounts) SetClientEmail(v *string) *ServiceAccounts {
+	if o.ClientEmail = v; o.ClientEmail == nil {
+		o.nullFields = append(o.nullFields, "ClientEmail")
+	}
+	return o
+}
+
+func (o *ServiceAccounts) SetClientId(v *string) *ServiceAccounts {
+	if o.ClientId = v; o.ClientId == nil {
+		o.nullFields = append(o.nullFields, "ClientId")
+	}
+	return o
+}
+
+func (o *ServiceAccounts) SetAuthUri(v *string) *ServiceAccounts {
+	if o.AuthUri = v; o.AuthUri == nil {
+		o.nullFields = append(o.nullFields, "AuthUri")
+	}
+	return o
+}
+
+func (o *ServiceAccounts) SetTokenUri(v *string) *ServiceAccounts {
+	if o.TokenUri = v; o.TokenUri == nil {
+		o.nullFields = append(o.nullFields, "TokenUri")
+	}
+	return o
+}
+
+func (o *ServiceAccounts) SetAuthProviderX509CertUrl(v *string) *ServiceAccounts {
+	if o.AuthProviderX509CertUrl = v; o.AuthProviderX509CertUrl == nil {
+		o.nullFields = append(o.nullFields, "AuthProviderX509CertUrl")
+	}
+	return o
+}
+
+func (o *ServiceAccounts) SetClientX509CertUrl(v *string) *ServiceAccounts {
+	if o.ClientX509CertUrl = v; o.ClientX509CertUrl == nil {
+		o.nullFields = append(o.nullFields, "ClientX509CertUrl")
+	}
+	return o
+}
+
+type SetServiceAccountsInput struct {
+	ServiceAccounts *ServiceAccounts `json:"serviceAccount,omitempty"`
+}
+type SetServiceAccountsOutput struct {
+	ServiceAccounts *ServiceAccounts `json:"serviceAccount,omitempty"`
```

Comment:
According to api document set credentials response is only status code and message

Reply:
Followed same implementation as in aws.